packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -911,6 +911,80 @@ class OneFrameImageStreamCompleter extends ImageStreamCompleter {
   }
 }
 
+/// An iterator which iterates over the frames of an image.
+abstract class ImageFrameIterator {
+  /// A default constructor to allow for subclasses.
+  const ImageFrameIterator();
```

Comment:
This class is 100% abstract. Can we make it an [interface class](https://dart.dev/language/class-modifiers#interface) and remove this constructor? Subtypes can implement it instead of extending.

Reply:
Done.